From f55d7e2050223fb7943891bc9976ee30b88b08a6 Mon Sep 17 00:00:00 2001
From: Wojciech Bartczak <wbartczak@marvell.com>
Date: Tue, 1 Jun 2021 05:06:59 -0700
Subject: [PATCH 1577/1921] driver: soc: Fix OcteonTX info driver

The driver uses information provided in device tree to present
board information to the user. The change fixes problem in
data interpretation for BOARD-MAC-ADDRESS-XXX-NUM.
The field is coded as hexadecimal string, while kernel driver
decodes this field as decimal string.

Signed-off-by: Wojciech Bartczak <wbartczak@marvell.com>
Change-Id: Idd4733d0284854683ce44a7e755dbc60b1ed80f9
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/53166
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/octeontx_info.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx_info.c b/drivers/soc/marvell/octeontx_info.c
index 62484f10d7e1..309c7cf66245 100644
--- a/drivers/soc/marvell/octeontx_info.c
+++ b/drivers/soc/marvell/octeontx_info.c
@@ -162,7 +162,7 @@ static int octtx_parse_mac_info(struct device_node *node)
 		brd.board_num_of_mac = null_string;
 		num = -1;
 	} else {
-		if (kstrtou32(brd.board_num_of_mac, 10, &num))
+		if (kstrtou32(brd.board_num_of_mac, 16, &num))
 			pr_warn("Board MAC address number is not available\n");
 	}
 
@@ -182,7 +182,7 @@ static int octtx_parse_mac_info(struct device_node *node)
 		brd.board_num_of_mac_id = null_string;
 		id_num = -1;
 	} else {
-		if (kstrtou32(brd.board_num_of_mac_id, 10, &id_num))
+		if (kstrtou32(brd.board_num_of_mac_id, 16, &id_num))
 			pr_warn("Board MAC addressess IDs number is not available\n");
 	}
 
-- 
2.31.1

